src/umpire/ResourceManager.cpp

```diff
@@ -316,10 +315,12 @@ void ResourceManager::copy(void* dst_ptr, void* src_ptr, size_t size)
   auto& op_registry = op::MemoryOperationRegistry::getInstance();
 
   auto src_alloc_record = m_allocations.find(src_ptr);
+  std::ptrdiff_t src_offset = static_cast<char*>(src_ptr) - static_cast<char*>(src_alloc_record->m_ptr);
+  std::size_t src_size = src_alloc_record->m_size - src_offset;
+  
   auto dst_alloc_record = m_allocations.find(dst_ptr);
-
-  std::size_t src_size = src_alloc_record->m_size;
-  std::size_t dst_size = dst_alloc_record->m_size;
+  std::ptrdiff_t dst_offset = static_cast<char*>(dst_ptr) - static_cast<char*>(dst_alloc_record->m_ptr);
+  std::size_t dst_size = dst_alloc_record->m_size - dst_offset;
```

Comment:
This looks good as it appears to fix an offset-related memory operation bug.  Is this considered a leak or a different bug?  Was this found from testing or inspection?

Reply:
This is a different bug, but worth fixing.

Reply:
I agree.  But did we should make sure that tests are added to cover this case if they did not exist before.

Reply:
@mcfadden8 the `memset` functionality is being exercised but within a different unit test. I'll add two new tests which specifically exercise the offset functionality.

Reply:
Done.